File: commons/test-utils/src/test/java/com/cedarsoft/test/utils/TemporaryFolderExtensionTest.java
package com.cedarsoft.test.utils;

import java.io.File;

import javax.annotation.Nonnull;

import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.*;

/**
 * @author Johannes Schneider (<a href="mailto:js@cedarsoft.com">js@cedarsoft.com</a>)
 */
@WithTempFiles
class TemporaryFolderExtensionTest {
  @Test
  void testTmpFolder(@Nonnull @TempFolder File folder) {
    Assertions.assertThat(folder).isNotNull().exists().isDirectory();
  }

  @Test
  void testTmpFile(@Nonnull @TempFile File file) {
    Assertions.assertThat(file).isNotNull().exists().isFile();
  }

  @Test
  void testTempFolder(@Nonnull TemporaryFolder tmp) throws Exception {
    Assertions.assertThat(tmp.newFolder()).exists().isDirectory();
  }
}

File: commons/test-utils/src/test/kotlin/it/neckar/open/test/utils/TemporaryFolderExtensionTest.kt
package it.neckar.open.test.utils

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.io.File

/**
 */
@WithTempFiles
internal class TemporaryFolderExtensionTest {
  @Test
  fun testTmpFolder(@TempFolder folder: File?) {
    assertThat(folder).isNotNull.exists().isDirectory
  }

  @Test
  fun testTmpFile(@TempFile file: File?) {
    assertThat(file).isNotNull.exists().isFile
  }

  @Test
  fun testTempFolder(tmp: TemporaryFolder) {
    assertThat(tmp.newFolder()).exists().isDirectory
  }
}
